blog/2024-08-19-gsoc-vertx-proxy-updates.mdx

```diff
@@ -0,0 +1,123 @@
+in the official release and how to use them, followed by reflections on the GSoC experience.
+
+<Alert warning>
+The updates mentioned in this post are not yet available from the maven repository; The actual results are subject 
+to the final provided API.
+</Alert>
+
+## Interceptors
+
+During message forwarding, developers can add interceptors to intercept and modify incoming requests and responses. 
+This update provides a quick way to create different types of interceptors. For example, now it's possible to create 
+a interceptor to filter headers with a single method:
+
+```java
+proxy.addInterceptor(
+    HeadersInterceptor.filterRequestHeaders(Set.of("header-to-remove")));
+```
+
+Here's a list of all the interceptors implemented:
+
+- `HeadersInterceptor`: Used to modify the headers of HTTP requests and responses;
+- `PathInterceptor`: used to modify the URI of the request;
+- `QueryInterceptor`: used to modify the query parameter of the request;
+- `BodyInterceptor`: used to modify the body of the request and response.
+
+Additionally, it is now possible to apply interceptors to WebSocket handshake packets. This means that you can now 
+modify the WebSocket request URL using an interceptor. Considering that in most cases you might not want 
+to apply all interceptor transformations during the WebSocket handshake (which could lead to connection failures), 
+interceptors are disabled for WebSocket by default. You can enable this behavior by overriding the `allowApplyToWebSocket` 
+method. For interceptors that have already been created, you can also achieve this by wrapping them with 
+`WebSocketInterceptor`:
+
+```java
+ProxyInterceptor addPrefix = PathInterceptor.addPrefix("/api");
+addPrefix = WebSocketInterceptor.allow(addPrefix); // now `addPrefix` can be applied to WebSocket	
+```
+
+## Cache Support 
+
+The proxy supports caching responses. In previous versions, caching was only allowed for responses with the `Cache-Control` 
+marked as `public`. This update improves the caching mechanism and meets most of the requirements of 
+[RFC9111](https://www.rfc-editor.org/rfc/rfc9111.html). The specific changes are as follows:
+
+- Implemented all cache request and response directives from RFC9111 (except `no-transform`);
+- By default, all responses are now allowed to be cached unless explicitly prohibited in the Cache-Control header;
+- More standard revalidation mechanism: expired caches will not be immediately deleted but will first be validated to determine if they can still be used;
+- Now revalidation supports both `ETag` and `Last-Modified` validators;
+- Fixed the issue where calling unsafe request methods would not invalidate the cache;
+- Fixed the issue with wrong caching responses when including the `Vary` header;
+- Implemented default heuristic freshness.
+
+A Cache SPI is also currently provided to support custom Cache implementations. A Cache instance can be shared by multiple proxies.
```

Comment:
```suggestion
A Cache SPI is also currently provided to support custom different Cache implementations (e.g. on heap or Redis).
```

I would not add the statement about cache sharing because it can be confusing without going into the details of how the cache works.